src/main/app/form/Form.ts

```diff
@@ -4,26 +4,13 @@ import { AnyObject } from '../controller/PostController';
 import { setupCheckboxParser } from './parser';
 
 export class Form {
-  constructor(private readonly form: FormContent, private formState: Partial<Case> = {}) {}
-
-  public setFormState(formState: Partial<Case>): void {
-    this.formState = formState;
-  }
-
-  public getFields(checkFields?: FormContent['fields']): FormContent['fields'] {
-    const fields = checkFields || this.form?.fields;
-    if (typeof fields === 'function') {
-      return fields(this.formState);
-    }
-
-    return fields;
-  }
+  constructor(private readonly fields: FormFields | FormFieldsFn) {}
 
   /**
    * Pass the form body to any fields with a parser and return mutated body;
    */
   public getParsedBody(body: AnyObject, checkFields?: FormContent['fields']): Partial<CaseWithFormData> {
-    const fields = this.getFields(checkFields);
+    const fields = checkFields || this.fields;
```

Comment:
is checkFields still needed? does anything use it?

Reply:
it's needed for the subfields

Reply:
.map(subField => this.getParsedBody(body, subField))